### PR TITLE
Spec: difference between forall- and []-loops

### PR DESCRIPTION
**Pull request by vasslitvinov**
Update the spec to indicate that forall-statements and expressions
require parallelism, whereas []-ones do not.

### CODE REVIEW

File: spec/Data_Parallelism.tex [hunk 1]
```diff
@@ -52,7 +52,10 @@ \subsection{Syntax}
 \end{syntax}
 As with the for statement, the indices may be omitted if they are
 unnecessary and the \chpl{do} keyword may be omitted before a block
-statement.  The square bracketed form is a syntactic convenience.
+statement.  The square bracketed form will resort to serial iteration
```
Comment (lydia-duncan) [inline spec/Data_Parallelism.tex]: I would make this a separate paragraph in both places.  I might still mention that the square bracketed form is for convenience, while still emphasizing its difference in behavior.

**Comment by vasslitvinov** [comment]
@lydia-duncan how does it look now?

I did not reintroduce "for convenience" because IMO that can mislead w.r.t. what kind of convenience it is.

**Comment by lydia-duncan** [comment]
My thinking is that it would be handled like "The square-bracket form may be used for convenience, but will resort to serial iteration ...", but I'm okay with this going in as is if that isn't appealing

**Comment by vasslitvinov** [comment]
Thanks Lydia!

**Pull request merged**